ts/Extensions/Annotations/Types/FibonacciTimezones.ts

```diff
@@ -0,0 +1,292 @@
+/* *
+ *
+ *  Author: Rafal Sebestjanski
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type PositionObject from '../../../Core/Renderer/PositionObject';
+
+import Annotation from '../Annotations.js';
+import ControlPoint from '../ControlPoint.js';
+import CrookedLine from './CrookedLine.js';
+import InfinityLine from './InfinityLine.js';
+import MockPoint from '../MockPoint.js';
+import U from '../../../Core/Utilities.js';
+const { merge } = U;
+
+/**
+ * Internal types.
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface AnnotationControllable {
+            secondLineEdgePoints: [Function, Function];
+        }
+    }
+}
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/* eslint-disable no-invalid-this, valid-jsdoc */
+
+class FibonacciTimezones extends CrookedLine {
+
+    /* *
+     *
+     * Static Functions
+     *
+     * */
+
+    private static edgePoint(
+        startIndex: number,
+        endIndex: number,
+        fibonacciIndex: number
+    ): Function {
+        return function (target: any): PositionObject {
+            let annotation = target.annotation,
+                points = annotation.points,
+                // Offset between the first and the second line
+                deltaX = points.length > 1 ? points[1].x - points[0].x : 0,
+                // firstLine.x + fibb * offset
+                x = points[0].x + fibonacciIndex * deltaX;
+
+            // Horizontal and vertical lines have only one point,
+            // make a copy of it:
+            points = [
+                new MockPoint(
+                    annotation.chart,
+                    points[0].target,
+                    {
+                        x: x,
+                        y: 0,
+                        xAxis: points[0].options.xAxis,
+                        yAxis: points[0].options.yAxis
+                    }
+                ),
+                new MockPoint(
+                    annotation.chart,
+                    points[0].target,
+                    {
+                        x: x,
+                        y: 1,
+                        xAxis: points[0].options.xAxis,
+                        yAxis: points[0].options.yAxis
+                    }
+                )
+            ];
+
+            return InfinityLine.findEdgePoint(
+                points[startIndex],
+                points[endIndex]
+            );
+        };
+    }
+
+    /* *
+     *
+     *  Constructors
+     *
+     * */
+
+    public constructor(chart: Highcharts.AnnotationChart, options: FibonacciTimezones.Options) {
+        super(chart, options);
+    }
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public addShapes(): void {
+        let fibb = 1,
+            nextFibb = 1,
+            temp: number;
+
+        for (let i = 0; i < 11; i++) {
```

Comment:
I had the same concern. I would leave it hardcoded for now and, when users want to change the number of lines (more than 11), then make it configurable.
What do you think?

I will add a comment for sure describing what `11` means.

Reply:
For me, it could be hardcoded the only thing I would change is a description of that property in case if in the future we might change it into something more configurable. ;)